crates/bevy_render/src/pipeline/vertex_buffer_descriptor.rs

```diff
@@ -1,69 +1,62 @@
 use super::VertexFormat;
-use bevy_utils::HashMap;
-use std::borrow::Cow;
-
-pub use bevy_derive::AsVertexBufferDescriptor;
-
-#[derive(Clone, Debug, Eq, PartialEq)]
+use bevy_property::Property;
+use serde::{Deserialize, Serialize};
+use std::{
+    borrow::Cow,
+    hash::{Hash, Hasher},
+};
+
+#[derive(Clone, Debug, Eq, PartialEq, Default, Property, Serialize, Deserialize)]
 pub struct VertexBufferDescriptor {
     pub name: Cow<'static, str>,
     pub stride: u64,
     pub step_mode: InputStepMode,
     pub attributes: Vec<VertexAttributeDescriptor>,
 }
 
+pub const VERTEX_FALLBACK_LAYOUT_NAME: &str = "Fallback";
 impl VertexBufferDescriptor {
-    pub fn sync_with_descriptor(&mut self, descriptor: &VertexBufferDescriptor) {
-        for attribute in self.attributes.iter_mut() {
-            let descriptor_attribute = descriptor
-                .attributes
-                .iter()
-                .find(|a| a.name == attribute.name)
-                .unwrap_or_else(|| {
-                    panic!(
-                        "Encountered unsupported Vertex Buffer Attribute: {}",
-                        attribute.name
-                    );
-                });
-            attribute.offset = descriptor_attribute.offset;
+    pub fn new_from_attribute(
+        attribute: VertexAttributeDescriptor,
+        step_mode: InputStepMode,
+    ) -> VertexBufferDescriptor {
+        VertexBufferDescriptor {
+            name: attribute.name.clone(),
+            stride: attribute.format.get_size(),
+            step_mode,
+            attributes: vec![attribute.clone()],
         }
+    }
 
-        self.stride = descriptor.stride;
+    // just for tests, since a reflected layout doesn't know about the stride
```

Comment:
not a huge fan of creating methods directly on structs for unit tests. can we define this as a helper within the unit test module?